net/boinc/Makefile

```diff
@@ -0,0 +1,93 @@
+# Copyright 2020 by Christian Dreihsig and Steffen Möller
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=boinc
+PKG_VERSION:=7.16.5
+PKG_VERSION_SHORT:=$(shell echo $(PKG_VERSION)| cut -f1,2 -d.)
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/BOINC/boinc
```

Comment:
Please use `codeload.github.com` instead of raw git. Grep in other packages to see how they do it.

Reply:
@aparcar , we tried that but somehow all URLs we came up with have failed. This is a bit mysterious. Could you possibly help us with that?

Reply:
Update: Done.

Reply:
It looks still the same to me. What did you change? If you still need help please let me know.

Reply:
@aparcar Yes, please. The first success was because of a redundant feed with the previous solution :o/ My bad. The only solution we managed to come up with did not allow to have the PKG_VERSION remain as it is (7.16.5). The URL https://codeload.github.com/BOINC/boinc/tar.gz/fbdcc2d6165f0a3897341852f0fcf91a934724d0 (as a .tar.gz, not repacked into a tar.xz that we would have preferred) would download the right tarball but nothing downstream would work since the directory untared would be boinc-<that long hash>, not boinc-7.16.5. BOINC distinguishes server and client releases from the same code base (https://github.com/BOINC/boinc/releases) which may cause the trouble?!? There does not seem to be much documentation on codeload.github.com out there. Many thanks for your offer to help us out. Much appreciated.  Steffen

Reply:
First tried telnet on port 80, then lynx - the URL BOINC redirects to is https://codeload.github.com/BOINC/boinc/tar.gz/client_release/7.16/7.16.5.  Got some use for the short version again .... are you completely confident that this is more elegant than what we have? I understand it is one web redirection less, but ... let alone our time from this eving is difficult to pay back with that increase in efficienty.

Reply:
I think it is fair to say that I am at my wits' end. The tarball downloaded does not untar as "boinc-7.16.5" but as "boinc-client_release-7.16-7.16.5". No idea how to undress that except for overloading the download target. This are the lines that brought me here
```
PKG_SOURCE_URL:=https://codeload.github.com/BOINC/boinc/tar.gz/client_release/7.16/$(PKG_VERSION)?
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_DATE:=2020-02-25
PKG_HASH:=33db60991b253e717c6124cce4750ae7729eaab4e54ec718b9e37f87012d668a
```
Many, many thanks for all you can possibly do to help us out.

Reply:
Please try the following. The build system is great and powerful, however it's perks not perfectly documented :/

```
PKG_VERSION_SHORT:=$(shell echo $(PKG_VERSION)| cut -f1,2 -d.)
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://codeload.github.com/BOINC/boinc/tar.gz/client_release/$(PKG_VERSION_SHORT)/$(PKG_VERSION)?
PKG_HASH:=33db60991b253e717c6124cce4750ae7729eaab4e54ec718b9e37f87012d668a

PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-client_release-$(PKG_VERSION_SHORT)-$(PKG_VERSION)
```

Reply:
This worked like a charm. Thank you again.